ros/src/computing/planning/motion/packages/op_local_planner/nodes/op_behavior_selector/op_behavior_selector_core.cpp

```diff
@@ -0,0 +1,495 @@
+				"time,dt, Behavior_i, Behavior_str, RollOuts_n, Blocked_i, Central_i, Selected_i, StopSign_id, Light_id, Stop_Dist, Follow_Dist, Follow_Vel,"
+				"Target_Vel, PID_Vel, T_cmd_Vel, C_cmd_Vel, Vel, Steer, X, Y, Z, Theta,"
+				, m_LogData);
+}
+
+void BehaviorGen::UpdatePlanningParams(ros::NodeHandle& _nh)
+{
+	_nh.getParam("/op_common_params/enableSwerving", m_PlanningParams.enableSwerving);
+	if(m_PlanningParams.enableSwerving)
+		m_PlanningParams.enableFollowing = true;
+	else
+		_nh.getParam("/op_common_params/enableFollowing", m_PlanningParams.enableFollowing);
+
+	_nh.getParam("/op_common_params/enableTrafficLightBehavior", m_PlanningParams.enableTrafficLightBehavior);
+	_nh.getParam("/op_common_params/enableStopSignBehavior", m_PlanningParams.enableStopSignBehavior);
+
+	_nh.getParam("/op_common_params/maxVelocity", m_PlanningParams.maxSpeed);
+	_nh.getParam("/op_common_params/minVelocity", m_PlanningParams.minSpeed);
+	_nh.getParam("/op_common_params/maxLocalPlanDistance", m_PlanningParams.microPlanDistance);
+
+	_nh.getParam("/op_common_params/pathDensity", m_PlanningParams.pathDensity);
+
+	_nh.getParam("/op_common_params/rollOutDensity", m_PlanningParams.rollOutDensity);
+	if(m_PlanningParams.enableSwerving)
+		_nh.getParam("/op_common_params/rollOutsNumber", m_PlanningParams.rollOutNumber);
+	else
+		m_PlanningParams.rollOutNumber = 0;
+
+	_nh.getParam("/op_common_params/horizonDistance", m_PlanningParams.horizonDistance);
+	_nh.getParam("/op_common_params/minFollowingDistance", m_PlanningParams.minFollowingDistance);
+	_nh.getParam("/op_common_params/minDistanceToAvoid", m_PlanningParams.minDistanceToAvoid);
+	_nh.getParam("/op_common_params/maxDistanceToAvoid", m_PlanningParams.maxDistanceToAvoid);
+	_nh.getParam("/op_common_params/speedProfileFactor", m_PlanningParams.speedProfileFactor);
+
+	_nh.getParam("/op_trajectory_evaluator/horizontalSafetyDistance", m_PlanningParams.horizontalSafetyDistancel);
+	_nh.getParam("/op_trajectory_evaluator/verticalSafetyDistance", m_PlanningParams.verticalSafetyDistance);
+
+	_nh.getParam("/op_common_params/enableLaneChange", m_PlanningParams.enableLaneChange);
+
+	_nh.getParam("/op_common_params/width", m_CarInfo.width);
+	_nh.getParam("/op_common_params/length", m_CarInfo.length);
+	_nh.getParam("/op_common_params/wheelBaseLength", m_CarInfo.wheel_base);
+	_nh.getParam("/op_common_params/turningRadius", m_CarInfo.turning_radius);
+	_nh.getParam("/op_common_params/maxSteerAngle", m_CarInfo.max_steer_angle);
+	_nh.getParam("/op_common_params/maxAcceleration", m_CarInfo.max_acceleration);
+	_nh.getParam("/op_common_params/maxDeceleration", m_CarInfo.max_deceleration);
+	m_CarInfo.max_speed_forward = m_PlanningParams.maxSpeed;
+	m_CarInfo.min_speed_forward = m_PlanningParams.minSpeed;
+
+	PlannerHNS::ControllerParams controlParams;
+	controlParams.Steering_Gain = PlannerHNS::PID_CONST(0.07, 0.02, 0.01);
+	controlParams.Velocity_Gain = PlannerHNS::PID_CONST(0.1, 0.005, 0.1);
```

Comment:
are these param generic and not vehicle specific? I know that they are just initial conditions but still

Reply:
yes, they should be tuned for every robot, and should be parameterized.